### PR TITLE
fix(metrics): Update indexer batch message metrics to not use average of averages

### LINKED ISSUE #61537: Indexer: update value length and message size metrics to use weighted average indead of average of average
Currently the indexer metrics for value length and message size works as follows:
For each batch create a mapping of:
- Use case id → value length
- Use case id → message size
- Use case id → number of messages seen

At the end of the batch, we divide the value length / # tag key for a use case with the number of messages for that use case we have seen. This works for tacking the metrics for each individual use case, but it doesn't work for tracking the total average because the actual average needs to be weighted.

### PR DESCRIPTION
**Pull request by john-z-yang**
### Overview

1. The indexer emit metrics such as "average message size per batch for each use case", by summing the message size for a use case (numerator) and dividing it by the number of message for that use case in a batch (denominator). This pr emits the the 2 values separately, so we can calculate the weighted average (i.e. the average message size for all messages per batch).

2. Update the value length metric to also include counters.

### CODE REVIEW

**Review by 50-Course — COMMENTED** [review]
By my understanding;

- The previous diff allows for `average message size per batch for each use case` at best. 

- Newer diff, seemed to match the objective of success criteria MR, of not explicitly calling the averages.

**Review by ayirr7 — APPROVED** [review]

**Comment by ayirr7** [comment]
To be clear, this PR converts every metric we were emitting here back into its raw/summed value (which we will then break down by specific stats like `avg`, `max`, etc.) afterwards?

File: src/sentry/sentry_metrics/consumers/indexer/batch.py [hunk 1]
```diff
@@ -182,23 +175,12 @@ def _extract_message(
...
-            len(msg.payload.value), self.__message_bytes_max[use_case_id]
-        )
-
-        self.__message_tags_len_sum[use_case_id] += len(parsed_payload.get("tags", {}))
-        self.__message_tags_len_max[use_case_id] = max(
-            len(parsed_payload.get("tags", {})), self.__message_tags_len_max[use_case_id]
+        self.__message_bytes[use_case_id] += len(msg.payload.value)
+        self.__message_tags_len[use_case_id] += len(parsed_payload.get("tags", {}))
+        self.__message_value_len[use_case_id] += (
+            len(parsed_payload["value"]) if isinstance(parsed_payload["value"], Iterable) else 1
```
Comment (ayirr7) [inline src/sentry/sentry_metrics/consumers/indexer/batch.py:181]: Thanks for this. Indeed, we want to still count the presence of other metric types (other than just `set` and `distribution`). One point to consider though, might be that do we want to use `1` universally as so? For counters, this makes sense. I wonder if it does for gauges as well?

Comment (john-z-yang) [reply src/sentry/sentry_metrics/consumers/indexer/batch.py:181]: Isn't the `value` field for gauges a `TypedDict`? Wouldn't that be an `Iterable`?

**Comment by john-z-yang** [comment]
> By my understanding;
> 
> * The previous diff allows for `average message size per batch for each use case` at best.
> * Newer diff, seemed to match the objective of success criteria MR, of not explicitly calling the averages.



> To be clear, this PR converts every metric we were emitting here back into its raw/summed value (which we will then break down by specific stats like `avg`, `max`, etc.) afterwards?

That's correct

File: src/sentry/sentry_metrics/consumers/indexer/batch.py (hunk 1 above)
Comment (ayirr7) [reply src/sentry/sentry_metrics/consumers/indexer/batch.py:181]: That's right, yep

**Pull request merged**